### ISSUE TITLE
Document how to custom MethodSecurityMetadataSource

### ISSUE THREAD

**Issue by quaff** [spring-projects/spring-security#12345]
```java
import java.lang.reflect.Method;
import java.util.Collection;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.access.ConfigAttribute;
import org.springframework.security.access.annotation.SecuredAnnotationSecurityMetadataSource;
import org.springframework.security.access.method.MethodSecurityMetadataSource;
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
import org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration;
import org.springframework.util.CollectionUtils;

@Configuration(proxyBeanMethods = false)
@EnableGlobalMethodSecurity(prePostEnabled = true, proxyTargetClass = true)
public class MethodSecurityConfiguration extends GlobalMethodSecurityConfiguration {

	@Override
	protected MethodSecurityMetadataSource customMethodSecurityMetadataSource() {
		// override @EnableGlobalMethodSecurity(securedEnabled=true)
		return new SecuredAnnotationSecurityMetadataSource() {

			@Override
			public Collection<ConfigAttribute> getAttributes(Method method, Class<?> targetClass) {
				Collection<ConfigAttribute> attr = super.getAttributes(method, targetClass);
				if (CollectionUtils.isEmpty(attr)) {
					attr = findAttributes(targetClass);
				}
				return attr;
			}

		};
	}

}

```
`GlobalMethodSecurityConfiguration` is deprecated since 5.8.0 as
```java
 * @deprecated Use {@link PrePostMethodSecurityConfiguration},
 * {@link SecuredMethodSecurityConfiguration}, or
 * {@link Jsr250MethodSecurityConfiguration} instead
```
please elaborate how to custom MethodSecurityMetadataSource.

**Comment by jzheaux** [comment]
Hi, @quaff, thanks for reaching out.

What is your custom code trying to do? It appears similar in concept to what [`SecuredAuthorizationManager` is doing](https://github.com/spring-projects/spring-security/blob/main/core/src/main/java/org/springframework/security/authorization/method/SecuredAuthorizationManager.java#L68), which is wired by doing `@EnableMethodSecurity(securedEnabled = true)`:

**Comment by quaff** [comment]
@jzheaux It is a workaround of https://github.com/spring-projects/spring-security/pull/10230

**Comment by jzheaux** [comment]
Thanks, @quaff.

`AuthorizationManager` does not have a separate API for customizing the source of security metadata. Instead, consider replacing your custom `MethodSecurityMetadataSource` with a custom `AuthorizationManager`.

In the meantime, you can achieve it by calling your `MethodSecurityMetadataSource` implementation like so:

```java
public class MyAuthorizationManager implements AuthorizationManager<MethodInvocation> {
    private final MethodSecurityMetadataSource metadata = // ... custom

    @Override 
    public AuthorizationDecision check(Supplier<Authentication> authentication, MethodInvocation mi) {
        Collection<String> attributes = toStringList(this.metadata.getAttributes(mi));
        Collection<String> authorities = toStringList(authentication.get().getAuthorities());
        return new AuthorizationDecision(!Collections.disjoint(attributes, authorities));
    }
}
```

And then publishing [the corresponding method interceptor](https://docs.spring.io/spring-security/reference/5.8/servlet/authorization/method-security.html#jc-method-security-custom-authorization-manager). You can represent your metadata source as a pointcut in the following way:

```java
Pointcut pointcut = new StaticMethodMatcherPointcut() {
    @Override 
    public boolean matches(Method method, Class<?> targetClass) {
        return !CollectionUtils.isEmpty(this.metadata.getAttributes(m, targetClass));
    } 
};
```

This process is indirectly documented [in the migration guide](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/authorization.html#_i_use_a_custom_accessdecisionvoter_2) however, I think that you are right that the documentation should speak directly to custom `SecurityMetadataSource` implementations.

**Comment by quaff** [comment]
1. `MethodSecurityMetadataSource` is deprecated also.
2. this way is not intuitive as previous.

@jzheaux Could you consider provide a separate API in next release?

**Comment by jzheaux** [comment]
> 1. `MethodSecurityMetadataSource` is deprecated also.

Correct, the point of the above snippet was to suggest how you could adapt your existing code. Certainly, you can write an `AuthenticationManager` from that looks up the values itself instead of delegating to your existing component.

> 2. this way is not intuitive as previous.

What parts are less intuitive? Do you think that some of this is being unfamiliar with the new API? The thing I'm speaking to specifically is we've gone from four interfaces (SecurityMetadataSource, AccessDecisionManager, AccessDecisionVoter, AfterInvocationManager) down to one (AuthorizationManager), which seems like it would be more intuitive.

**Comment by quaff** [comment]
> What parts are less intuitive? I want to make sure we are simplifying the right thing.

Overriding method `customMethodSecurityMetadataSource` is more intuitive.

> It's worth considering, sure. I'm inclined to see what patterns emerge first before adding more abstractions.

Thanks for your considering.
